packages/sdk/src/conversation/middlewares/commandMiddleware.ts

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Activity, ActivityTypes, Middleware, TurnContext } from "botbuilder";
+import { CommandMessage, TeamsFxBotCommandHandler, TriggerPatterns } from "../interface";
+
+/**
+ * @internal
+ */
+export class CommandResponseMiddleware implements Middleware {
+  public readonly commandHandlers: TeamsFxBotCommandHandler[] = [];
+
+  constructor(handlers?: TeamsFxBotCommandHandler[]) {
+    if (handlers && handlers.length > 0) {
+      this.commandHandlers.push(...handlers);
+    }
+  }
+
+  public async onTurn(context: TurnContext, next: () => Promise<void>): Promise<void> {
+    if (context.activity.type === ActivityTypes.Message) {
+      // Invoke corresponding command handler for the command response
+      const commandText = this.getActivityText(context.activity);
+
+      const message: CommandMessage = {
+        text: commandText,
+      };
+
+      for (const handler of this.commandHandlers) {
+        const matchResult = this.shouldTrigger(handler.triggerPatterns, commandText);
+
+        // It is important to note that the command bot will stop processing handlers
+        // when the first command handler is matched.
```

Comment:
Although this is the old code but just confused... how do we ensure only one handler is triggered? There's no `break` in the for-each iteration.

Reply:
Nice catch! we should add `break` here to stop processing if one handler is matched.